### ISSUE TITLE
Create UserProfile endpoint

### ISSUE THREAD

**Issue by MaksymGavatuk** [BinaryStudioAcademy/bsa-2023-bsa-talents#172]
Create endpoint to add user profile information. Flow will be the next - we will save user in every registration step. So finish first step => call PATCH method => finish second step => push PATCH method...
- [x] create user details repository
- [x] create user details service
- [x] create user details controller
- [x] create dto types for endpoint 
- [x] add endpoint (PATCH - method)
- [x] add validation for this method

**Comment by nikita-remeslov** [comment]
@MaksymGavatuk let's split this task

POST
PATCH
etc.

**Comment by JosuerBague** [comment]
I have draft branch on this task. I will push to remote. 

I agree task needs to be split if we intend to use it this Friday.

**Comment by JosuerBague** [comment]
I was thinking maybe when a new user is created, a new user details entity is created automatically but with null values for the fields. (Other than userId)

**Comment by JosuerBague** [comment]
What I've done so far:

1) User Details Repository with working create, update, and find.
2) User Details Service with working create, update, and find.
3) User Details controller (paths are unsure).

I've added this line to auth.controller.signUp:
```
 private async signUp(
        options: ApiHandlerOptions<{
            body: UserSignUpRequestDto;
        }>,
    ): Promise<ApiHandlerResponse> {
        const user = await this.authService.signUp(options.body);
        await this.userDetailsService.create({ id: user.id });    // Creates empty details entity in db.
        return {
            status: HttpCode.CREATED,
            payload: user,
        };
    }
```

My line of reasoning is this.
1) User signs up.
2) Empty details entity is created in db which will be updated as user goes through the onboarding steps.



![Image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/8c9697f6-da70-4629-b2b5-72fbf511d52f)


The one entity with an 'is_approved' true, is the result of me testing userDetailsService.update(). But I did not test this through userDetailsService controller yet.
